autorest/adal/token.go

```diff
@@ -339,6 +355,35 @@ func NewServicePrincipalTokenFromCertificate(oauthConfig OAuthConfig, clientID s
 	)
 }
 
+// NewServicePrincipalTokenFromUsernamePassword create a ServicePrincipalToken from the username and password
```

Comment:
Nit, please add this link (documentation) to the comment
https://blogs.msdn.microsoft.com/wushuai/2016/09/25/resource-owner-password-credentials-grant-in-azure-ad-oauth/